src/types/account-memberships.d.ts

```diff
@@ -1,64 +1,85 @@
 /**
  * Account Memberships
  */
-import {Resource,} from './core'
+import { Resource, ResourceList } from './core'
 
 /**
  * The Account Membership object Interface
  */
 
 export interface AccountMembership {
-    type: string
-    meta: {
-        timestamps: {
-            created_at: string,
-            updated_at: string
-        }
-    },
-    relationships: {
-        "account-member": {
-            data: {
-                id: string,
-                type: string
-            }
-        }
+  type: string
+  meta: {
+    timestamps: {
+      created_at: string
+      updated_at: string
     }
+  }
+  relationships: {
+    'account-member': {
+      data: {
+        id: string
+        type: string
+      }
+    }
+  }
 }
 
 export interface AccountMembershipCreateBody {
-    type: string,
-    account_member_id: string
+  type: string
+  account_member_id: string
 }
 
 /**
  * Account Memberships Endpoints
  */
 export interface AccountMembershipsEndpoint {
-    endpoint: 'account-member'
-    storage: Storage
+  endpoint: 'account-member'
+  storage: Storage
 
-    /**
-     * Get an Account Membership by reference
-     * @param accountId - The ID for the requested account,
-     * @param accountMembershipId - The ID for the requested account membership,
-     * @param token - The Bearer token to grant access to the API.
-     */
-    Get(accountId: string, accountMembershipId: string, token?: string): Promise<Resource<AccountMembership>>
+  /**
+   * Get an Account Membership by reference
+   * @param accountId - The ID for the requested account,
+   * @param accountMembershipId - The ID for the requested account membership,
+   * @param token - The Bearer token to grant access to the API.
+   */
+  Get(
+    accountId: string,
+    accountMembershipId: string,
+    token?: string
+  ): Promise<Resource<AccountMembership>>
 
-    /**
-     * Create an Account Membership
-     * @param accountId - The ID for the requested account,
-     * @param body
-     * @param token
-     */
-    Create(accountId: string, body: AccountMembershipCreateBody, token?: string): Promise<Resource<AccountMembership>>
+  /**
+   * Get all Account Memberships for an account
+   * @param accountId - The ID for the account
+   * @param token - The Bearer token to grant access to the API
+   */
+  All(
+    accountId: string,
+    token?: string
+  ): Promise<ResourceList<AccountMembership>>
```

Comment:
What happens to our endpoints when we add filtering. How would we manage adding those parameters in a way that is safe and not disrupting to js-sdk users. Maybe we need to add a note here as well.

Reply:
I assume that ResourceList just abstracts over our pagination rules.

Reply:
If we support filtering we should implement it as methods Limit, Offset methods, etc. You can get this by inheriting from `QueryableResource`. Take a look at `CurrencyEndpoint` for instance.